appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/impl/WeightedSnapshot.java

```diff
@@ -94,7 +94,7 @@ public WeightedSnapshot(Collection<WeightedSample> values) {
 
         for (int i = 0; i < copy.length; i++) {
             this.values[i] = copy[i].value;
-            this.normWeights[i] = copy[i].weight / sumWeight;
+            this.normWeights[i] = sumWeight == 0d ? 0d : copy[i].weight / sumWeight;
```

Comment:
from the context it should not be happening that `sumWeight` is zero but theoretical it could in case of bad test data. I figured that a normalised value of zero is the right value in case sum is zero.